internal/server.go

```diff
@@ -79,7 +79,7 @@ func NewServer(config *Config) (http.Handler, error) {
 
 	mux := http.NewServeMux()
 	chain := func(next http.Handler) http.Handler {
-		return POSTHandler(HeaderHandler(fleetLockHeaderKey, "true", next))
+		return InstrumentedHandler(s.metrics.totalRequests, s.metrics.responseStatus, s.metrics.httpDuration, POSTHandler(HeaderHandler(fleetLockHeaderKey, "true", next)))
```

Comment:
`InstrumentHandler` uses plenty of server state, you might as well make it a method and avoid passing all these values.

Reply:
Moved.